### PR TITLE
Moved Panache's JandexUtil to core's JandexUtil and AsmUtil

### PR DESCRIPTION
**Pull request by FroMage**
As discussed on quarkus-dev, required for #7733 and #9487

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkus/deployment/util/JandexUtil.java [hunk 1]
```diff
@@ -19,9 +20,14 @@
...
 
+import io.quarkus.builder.BuildException;
+
+/**
+ * A collection of Jandex utility methods.
+ */
 public final class JandexUtil {
 
-    private static final DotName OBJECT = DotName.createSimple(Object.class.getName());
+    public final static DotName DOTNAME_OBJECT = DotName.createSimple(Object.class.getName());
```
Comment (mkouba) [inline core/deployment/src/main/java/io/quarkus/deployment/util/JandexUtil.java:30]: I wonder why `org.jboss.jandex.DotName.OBJECT_NAME` is not public...

**Review by mkouba — APPROVED** [review]
Looks good.

File: core/deployment/src/main/java/io/quarkus/deployment/util/JandexUtil.java (hunk 1 above)
Comment (FroMage) [reply core/deployment/src/main/java/io/quarkus/deployment/util/JandexUtil.java:30]: Very good question. Perhaps @n1hility knows?

**Comment by mkouba** [comment]
There is a conflict in `JandexUtil` after I merged https://github.com/quarkusio/quarkus/pull/9637.

**Comment by FroMage** [comment]
Rebasing made the conflict go away, I had already backported the same fix to the new JandexUtil location.

**Pull request merged**